### ISSUE TITLE
Detect redundant collection type hints for `*args`, `**kwargs`

### ISSUE THREAD

**Issue by scop** [PyCQA/pylint#4880]
### Current problem

Clean up `*args` and `**kwargs` in function parameters from redundant collection type hints.

### Desired solution

Have pylint warn if:
- `*args` is annotated as a tuple of some kind: it's redundant as only the type of the `args` items is needed, and if
- `**kwargs` is hinted to be a dict of some kind; it's redundant as only the type of the `kwargs` values is needed

So instead of for example:
```python
def foo(*args: Tuple[Any], **kwargs: Dict[str, int]):
```
...it would be desirable to write:
```python
def foo(*args: Any, **kwargs: int):
```

### Additional context

Suggestions welcome if there are opinions that some other linter (flake8-something?) would be a better one to have this check in rather than pylint.

**Comment by Pierre-Sassoulas** [comment]
I think this checker is infeasible as it's hard to guess if there really is a ``Tuple`` in ``*args`` or a ``Dict[str, Any]`` in ``**kwargs``. Consider this for example:
```python
def foo(a: Tuple[Any], b: Dict[str, Any]):
    pass
 
def wrapfoo(*args: Tuple[Any], **kwargs: Dict[str, Any]):
    foo(*args, **kwargs)

wrapfoo(("1",), b={"c": 2})
```

**Comment by scop** [comment]
Oh, excellent point, failed entirely to think of that. And besides, at least mypy will already error out for cases like:
```python
def foo(*args: tuple) -> None:
    pass

foo("x")
```

```shellsession
t.py:4: error: Argument 1 to "foo" has incompatible type "str"; expected "Tuple[Any, ...]"
```

...so there's no need for other linters to check for this. Thanks for the sanity check and feeding it to me gently! :)
